src/http.cc

```diff
@@ -439,9 +439,12 @@ Response Request::execute() {
 
     requests.fdset(&fdread, &fdwrite, &fdexcep, &maxfd);
 
-    if (select(maxfd + 1, &fdread, &fdwrite, &fdexcep, nullptr) < 0) {
+    int select_res = select(maxfd + 1, &fdread, &fdwrite, &fdexcep, timeout_val);
```

Comment:
This does not work practically. If `curl` tool or `libcurl` hangs in MS Windows, we cannot do anything.

Reply:
Well, if so, please ignore this PR.

Reply:
> Well, if so, please ignore this PR.

Do you have a solution for this issue on Windows? I'm experiencing the same problem